src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEAFULT_SIZE = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    int size;
+
+    private Entry<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Entry[DEAFULT_SIZE];
+        size = 0;
+    }
+
+    private static class Entry<K, V> {
```

Comment:
The 'Entry' class should be parameterized with the same key and value types as the enclosing 'MyHashMap' to ensure type safety.